_posts/2025-05-08-opensearch-performance-3.0.md

```diff
@@ -0,0 +1,1013 @@
+    <tr>
+      <!-- No cell here because of rowspan from header row -->
+      <td class="left">Range queries</td>
+      <td>26.08</td>
+      <td>23.12</td>
+      <td>16.91</td>
+      <td>18.71</td>
+      <td>17.33</td>
+      <td>17.39</td>
+      <td>18.51</td>
+      <td>3.17</td>
+      <td>3.17</td>
+      <td class="light-green-clr">3.55</td>
+      <td class="light-green-clr">3.67</td>
+      <td class="green-clr">2.68</td>
+    </tr>
+    <tr>
+      <!-- No cell here because of rowspan from header row -->
+      <td class="left">Date histograms</td>
+      <td>6068</td>
+      <td>5249</td>
+      <td>5168</td>
+      <td>469</td>
+      <td>357</td>
+      <td>146</td>
+      <td>157</td>
+      <td>164</td>
+      <td>160</td>
+      <td class="light-green-clr">163.58</td>
+      <td class="light-green-clr">159.57</td>
+      <td class="green-clr">85.21</td>
+    </tr>
+    <tr>
+      <td colspan="2" class="bold left gray-clr">Aggregate (geo mean)</td>
+      <td>159.04</td>
+      <td>154.59</td>
+      <td>130.9</td>
+      <td>74.85</td>
+      <td>51.84</td>
+      <td>43.44</td>
+      <td>37.07</td>
+      <td>24.66</td>
+      <td>24.63</td>
+      <td class="light-green-clr">27.04</td>
+      <td class="light-green-clr">21.21</td>
+      <td class="green-clr">16.04</td>
+    </tr>
+    <tr>
+      <td colspan="2" class="bold left gray-clr">Speedup factor, compared to OS 1.3 (geo mean)</td>
+      <td>N/A</td>
+      <td>1.03</td>
+      <td>1.21</td>
+      <td>2.12</td>
+      <td>3.07</td>
+      <td>3.66</td>
+      <td>4.29</td>
+      <td>6.45</td>
+      <td>6.46</td>
+      <td class="light-green-clr">5.88</td>
+      <td class="light-green-clr">7.50</td>
+      <td class="green-clr">9.92</td>
+    </tr>
+    <tr>
+      <td colspan="2" class="bold left gray-clr">Relative latency, compared to OS 1.3 (geo mean)</td>
+      <td>N/A</td>
+      <td>97.20</td>
+      <td>82.31</td>
+      <td>47.06</td>
+      <td>32.60</td>
+      <td>27.31</td>
+      <td>23.31</td>
+      <td>15.51</td>
+      <td>15.49</td>
+      <td class="light-green-clr">17.00</td>
+      <td class="light-green-clr">13.34</td>
+      <td class="green-clr">10.09</td>
+    </tr>
+    </tbody>
+  </table>
+
+*Row 7 is Big 5 benchmark mean latency (ms) by OpenSearch version. Lower is better.* 
+*Row 8 and 9 are speedup factor and latency relative to 1.3 baseline.*
+
+The OpenSearch 3.0 release brings search performance gains from two main areas: 
+
+1. **Out-of-the-box improvements from Lucene 10**: Lucene 10 introduces hardware-level enhancements such as SIMD-accelerated vector scoring and more efficient postings decoding, which immediately benefit query performance with no configuration changes required. For example, minute-level date histogram aggregations run up to **60% faster**, `search_after` queries improve by **20--24%**, and scroll queries show modest latency reductions because of low-level enhancements in both Lucene 10 and JDK 21.
```

Comment:
```suggestion
1. **Out-of-the-box improvements in Lucene 10**: Lucene 10 introduces hardware-level enhancements such as SIMD-accelerated vector scoring and more efficient postings decoding, which immediately benefit query performance and require no configuration changes. For example, minute-level date histogram aggregations run up to **60% faster**, `search_after` queries improve by **20--24%**, and scroll queries show modest latency reductions because of low-level enhancements in both Lucene 10 and JDK 21.
```